Ingress/controllers/gce/backends.go

```diff
@@ -40,7 +40,17 @@ func portKey(port int64) string {
 }
 
 func beName(port int64) string {
-	return fmt.Sprintf("%v-%d", backendPrefix, port)
+	// TODO: Pipe the clusterName through, for now it saves code churn to just
+	// grab it globally, especially since we haven't decided how to handle
+	// namespace conflicts in the Ubernetes context.
+	if *clusterName == "" {
+		return fmt.Sprintf("%v-%d", backendPrefix, port)
+	}
+	taggedName := fmt.Sprintf("%v-%d%v%v", backendPrefix, port, clusterNameDelimiter, *clusterName)
+	if len(taggedName) > nameLenLimit {
+		taggedName = taggedName[:nameLenLimit]
+	}
+	return taggedName
```

Comment:
See comment below: I thought this was the actual cluster name retrieved via the apiserver/some other authoritative channel. If it's an arbitrary cluster-uid, this seems reasonable, at least for testing purposes.

That being said, if the cluster name is being passed through on creation to the kube-system pod in GKE/GCE, we probably still want something better than truncation. If we only truncate there is no easy way to determine whether a particular backend was created by `my-test-kubernetes-cluster1` or `my-test-kubernetes-cluster2`.